cmake/pybind11_catkin.cmake.in

```diff
@@ -1,30 +1,17 @@
 cmake_minimum_required(VERSION 2.8.3)
 
+# Configure pybind11 using the cmake file provided by the upstream package.
+# This finds python includes and libs and defines pybind11_add_module()
 list(APPEND CMAKE_MODULE_PATH "${CMAKE_CURRENT_LIST_DIR}")
-
-set(TEMP_PYTHON_EXECUTABLE ${PYTHON_EXECUTABLE})
-if(DEFINED PYBIND_PYTHON_EXECUTABLE)
-    set(PYTHON_EXECUTABLE ${PYBIND_PYTHON_EXECUTABLE} CACHE INTERNAL "")
-endif()
-
+set(PYBIND11_PYTHON_VERSION ${PYTHON_VERSION_STRING})
 include(pybind11Tools)
 
-# Cache variables so pybind11_add_module can be used in parent projects
-set(PYBIND11_INCLUDE_DIR ${PYBIND11_INCLUDE_DIR} CACHE INTERNAL "")
-set(PYTHON_LIBRARIES ${PYTHON_LIBRARIES} CACHE INTERNAL "")
-
-set(PYTHON_EXECUTABLE ${TEMP_PYTHON_EXECUTABLE} CACHE INTERNAL "")
+# set variables used by pybind11_add_module()
+set(PYBIND11_INCLUDE_DIR ${pybind11_catkin_INCLUDE_DIRS})
```

Comment:
For this PR to work for me, I had to change this line to 

```cmake
set(PYBIND11_INCLUDE_DIR ${pybind11_catkin_INCLUDE_DIRS}/pybind11_catkin)
```

Reply:
This should not be necessary as `pybind11_catkin_INCLUDE_DIRS` should already include `pybind11_catkin`. I will validate this with an install-space setup again...

Reply:
My setup is a separate install space, and then a devel space that extends the install space. In this case, the `pybind11_catkin_INCLUDE_DIRS` did not include `pybind11_catkin`.

Reply:
Indeed, in an install-space setup this fails. I will have a look again. Sorry for the noise.

Reply:
Should be fixed now. @wxmerkt, could you try again please?

Reply:
Thank you for the quick fix. This works now well for me. If you are happy, I would merge this PR and make a new bloom release to melodic and kinetic

Reply:
That would be great. Thanks.